controllers/nutanixmachine_controller.go

```diff
@@ -629,6 +604,63 @@ func (r *NutanixMachineReconciler) getOrCreateVM(rctx *nctx.MachineContext) (*nu
 		return nil, err
 	}
 
+	// if BootstrapImageName is configured
+	if rctx.NutanixMachine.Spec.BootstrapImageName != nil {
+		bootstrapImgName := *rctx.NutanixMachine.Spec.BootstrapImageName
+		imgUUID, err := GetImageUUID(rctx.Context, rctx.NutanixClient, &bootstrapImgName, nil)
+		if err != nil {
+			errMsg := fmt.Errorf("failed to get the bootstrap image with name %s. %v", bootstrapImgName, err)
+			rctx.SetFailureStatus(capierrors.CreateMachineError, errMsg)
+			return nil, err
+		}
+
+		bootstrapDisk := &nutanixClientV3.VMDisk{
+			DeviceProperties: &nutanixClientV3.VMDiskDeviceProperties{
+				DeviceType: utils.StringPtr("CDROM"),
+				DiskAddress: &nutanixClientV3.DiskAddress{
+					AdapterType: utils.StringPtr("IDE"),
+					DeviceIndex: utils.Int64Ptr(0),
+				},
+			},
+			DataSourceReference: &nutanixClientV3.Reference{
+				Kind: utils.StringPtr("image"),
+				UUID: &imgUUID,
+			},
+		}
+		diskList = append(diskList, bootstrapDisk)
+		vmSpec.Resources.DiskList = diskList
+
+		log.Info(fmt.Sprintf("Use the bootstrap image %s to create the VM %s.", bootstrapImgName, vmName))
+	} else {
```

Comment:
Is it enough to have one or the other? How is cluster specific data (certs, additional `preKubeadmCommands` and `files`) can be provided here?